### PR TITLE
Fix late join menu to handle disabled AI

### PR DESCRIPTION
**Pull request by cakexensen**
## About The Pull Request

This PR fixes the problem found in  #48916 when AI is disabled. The late join menu was originally retrieving the category name and color from the first job in each category (typically the department head) and using that to style the menu. When AI is disabled it couldn't provide this information, and would crash the late join menu during rendering. This has been fixed by storing the category color in `exp_jobsmap`, which is an existing global list of job categories that was appropriately structured to store this data. Additionally `exp_jobsmap` has been reordered to retain the category order of the late join screen. Existing uses of this list did not care about index ordering, and are thus unaffected.

## Why It's Good For The Game

This change allows /tg/station and downstream stations to disable the AI without breaking the late join menu, which would prevent anyone from joining after the round starts. It does not change the perceived behavior of the screen at all and should go unnoticed by players.

## Changelog
:cl:
fix: fixed late join menu to work when AI is disabled
/:cl:

### CODE REVIEW

**Comment by flufflycthu1u** [comment]
This should also fix issue #46956.

File: code/modules/jobs/jobs.dm [hunk 1]
```diff
@@ -60,16 +60,17 @@ GLOBAL_LIST_INIT(nonhuman_positions, list(
 	"Cyborg",
 	ROLE_PAI))
 
+// include department colors for rendering the late join menu
 GLOBAL_LIST_INIT(exp_jobsmap, list(
```
Comment (AnturK) [inline code/modules/jobs/jobs.dm]: I don't feel this is appropriate global to use. 
New one would be more fitting.
(Or ideally we'd have departments defined with types somewhere but that's bigger refactor)

File: code/modules/mob/dead/new_player/new_player.dm [hunk 2]
```diff
@@ -438,12 +438,14 @@
 			SSjob.prioritized_jobs -= prioritized_job
 	dat += "<table><tr><td valign='top'>"
 	var/column_counter = 0
-	for(var/list/category in list(GLOB.command_positions) + list(GLOB.engineering_positions) + list(GLOB.supply_positions) + list(GLOB.nonhuman_positions - "pAI") + list(GLOB.civilian_positions) + list(GLOB.medical_positions) + list(GLOB.science_positions) + list(GLOB.security_positions))
```
Comment (AnturK) [inline code/modules/mob/dead/new_player/new_player.dm]: You're also changing the list used by silicons here.

**Review by AnturK — CHANGES_REQUESTED** [review]
Let's not mix purposes here. Exp table is not job listing.

File: code/modules/jobs/jobs.dm (hunk 1 above)
Comment (danxensen) [reply code/modules/jobs/jobs.dm]: Moved to a new global.

File: code/modules/mob/dead/new_player/new_player.dm (hunk 2 above)
Comment (danxensen) [reply code/modules/mob/dead/new_player/new_player.dm]: Changed the silicon category to use the same list definition that was previously used.

File: code/modules/jobs/jobs.dm [hunk 3]
```diff
@@ -60,6 +60,18 @@ GLOBAL_LIST_INIT(nonhuman_positions, list(
 	"Cyborg",
 	ROLE_PAI))
 
+// job categories for rendering the late join menu
+GLOBAL_LIST_INIT(position_categories, list(
+	EXP_TYPE_COMMAND = list("jobs" = command_positions, "color" = "#ccccff"),
+	EXP_TYPE_ENGINEERING = list("jobs" = engineering_positions, "color" = "#ffeeaa"),
+	EXP_TYPE_SUPPLY = list("jobs" = supply_positions, "color" = "#ddddff"),
+	EXP_TYPE_SILICON = list("jobs" = GLOB.nonhuman_positions - "pAI", "color" = "#ccffcc"),
```
Comment (AnturK) [inline code/modules/jobs/jobs.dm]: Chop the glob here. Since it's all in context of global ss anyway.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/jobs/jobs.dm (hunk 3 above)
Comment (danxensen) [reply code/modules/jobs/jobs.dm]: Whoops, good ol copypastes. Fixed.

**Pull request merged**